src/Tools/dotnet-watch/test/MsBuildFileSetFactoryTest.cs

```diff
@@ -276,6 +277,53 @@ public async Task ProjectReferences_Graph()
             );
         }
 
+        [Fact]
+        public async Task ProjectReferences_RazorFiles()
+        {
+            // Arrange
+            var razorSdkTargetsPath = GetType().Assembly.GetCustomAttributes<AssemblyMetadataAttribute>()
+                .First(f => f.Key == "RazorSdkPath").Value;
+
+            _tempDir
+                .SubDir("src")
+                    .SubDir("MyLibrary")
+                        .WithCSharpProject("MyLibrary", out var proj3, sdk: "Microsoft.NET.Sdk.Razor")
+                        .WithProperty("CustomCollectWatchItems", "$(CustomCollectWatchItems);_RazorSdkCustomCollectWatchItems")
```

Comment:
We cannot reference the Sdk.props in this repo (same deal as https://github.com/dotnet/aspnetcore/issues/24278). Changing this in the project instead to simulate this.